src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -86,15 +88,18 @@ public FunctionDescriptor GetHttpFunctionOrNull(Uri uri)
         protected override void OnInitializeConfig(JobHostConfiguration config)
         {
             base.OnInitializeConfig(config);
-
+            _metricsLogger = new WebHostMetricsLogger();
             // Add our WebHost specific services
-            config.AddService<IMetricsLogger>(new WebHostMetricsLogger());
+            config.AddService<IMetricsLogger>(_metricsLogger);
         }
 
         protected override void OnHostStarted()
         {
             base.OnHostStarted();
-
+            if (_metricsLogger != null)
+            {
+                _metricsLogger.OnHostStartedEvent(Instance);
```

Comment:
Yeah that's right - it happens any time functions are added/removed, function.json metadata files change, etc.

How will this event data be used? I heard at one time that we wanted an even on a daily basis or something.